First draft reviews (#12)
bcgov/platform-developer-docs: setup-artifactory-service-account-repository-project.md

`oc delete ArtifactoryServiceAccount default`

Archeobot detects that the object has been deleted and removes the service account from Artifactory.

Then, the project provisioner detects the missing ArtifactoryServiceAccount object and creates a new one in your `tools` namespace, also called `default`. This happens within about 5 minutes. Archeobot detects the `new` object and creates a new service account for you in Artifactory. The username will be different. The random string at the end changes to reflect that the account is new with a new password and new privileges.

If you've accidentally deleted secrets for a different Artifactory service account (one you created yourself, but not the `default` on in your `tools` namespace), follow the same process. The project provisioner doesn't recreate the object for you, you need to do that yourself. Delete the object, wait for Archeobot to clean everything up, and then create a new ArtifactoryServiceAccount object. You can use the same ASA name but remember the username for the account is different, because it has a different random string at the end.

## Setup a project

> Once you have your Artifactory project you can add repositories and users, adjust roles, and check the results of Xray scans on artifacts.

Reviewer: ```suggestion
Once you have your Artifactory project you can add repositories and users, adjust roles, and check the results of Xray scans on artifacts.

**Note: In order to use an existing Artifactory service account with your new Artifactory project, you must add that service account to the project. There are instructions for this below. If you created the `ArtifactoryProject` object in the same Openshift namespace as an existing `ArtifactoryServiceAccount` object, the related service account is NOT automatically provided access to the project.**
```

### Add users and service accounts to a project

Once you've gotten your project, make sure to add your own service account to the project, as it's not added automatically when the project is created.

To add a new user account to your project, do the following:
1. Click the gear at the top of the menu on the left, then choose **Identity and Access Members**. Click **Add Member**.
2. On the **Users** tab, search for a username.
3. Add a user with their IDIR or GitHub ID. The user has to have already signed into Artifactory's GUI at least once. If the user hasn't done so yet, tell them to.

You can also add any Artifactory service account and select multiple users to add them all at once. Then, specify the role you want to grant them.
* **Contributor**: Service accounts are commonly given this role.
* **Developer**: Assign this role if you want a service account to manage artifacts.
* **Administrator**: Assign this role if you want a service account to manage access to the project.

You can also add additional roles to the project, if you want more finely-tuned control over who gets access to what.

> ### Create a project
> Use Artifactory Projects to give teams access to private repositories. If you want a private repository where you can push your own images or other artifacts, you'll need a project. If you don't want to be able to push your own images or other artifacts, you don't need a project.
> 
> You can create an ArtifactoryProject object on your OpenShift namespace with the following command:
> 
> `oc process -f https://raw.githubusercontent.com/bcgov/platform-services-archeobot/master/archeobot/config/samples/tmpl-artifactoryproject.yaml -p NAME="[APname]" | oc create -f -`
> 
> The `APname` won't completely reflect the name of the Project as it appears in Artifactory. The project name in Artifactory looks something like `[NamespacePlate]-[APname]`. For example, if you create an ArtifactoryProject object in the `a1b2c3-tools` namespace and give it an `APname` of `test`, then the Project's full name is `a1b2c3-test`.
> 
> The project is also given a shortname, called a project key. This is created automatically using the first letter of your `APname` and three digits of your namespace name. Using the previous example, the project key is `ta1b`. The project key must be unique.
> 
> When Archeobot sees a new ArtifactoryProject in your namespace, it sends the Platform Team a note that you want a new Project, which the team must approve. While you wait for your new ArtifactoryProject to be approved, you can look at it using the following command:
> 
> `oc describe artproj [APname]`.
> 
> In `spec`, there's an entry for your project key and an `approval_status` box that should show **pending**. This means that we haven't approved or rejected your request for an Artifactory project.
> 
> When the project is approved, the `approval_status` box should show **approved**. In this case, Archeobot knows that your ArtifactoryProject is approved, but it hasn't yet made the Project for you in Artifactory.
> 
> You may also see **nothing-to-approve**. This status is meant to say that there are no outstanding changes to be made. If you see this, it means that your project is created in Artifactory, and you can [sign in](https://artifacts.developer.gov.bc.ca).
> 
> **Note**: While you are able to patch the `approval_status` box, it doesn't mean you can approve your own project. The box is there for informational purposes. If you change it, Archeobot changes it back.
> 
> ### Rejected requests for projects or quota increases
> If you've made a request for a new Artifactory project or quota increase and it was rejected, reach out to the Platform Services team to ask why. If you think your request was rejected in error, let the team know. If the request should have been approved, they'll switch the status to `approved` and you'll see the changes applied.
> 
> If the team maintains the rejection, make sure you acknowledge the rejection. If you don't, further change requests are ignored. Acknowledge the rejection by changing your Artifactory projectProject object back to the state it was in before you made the request.
> * If you requested a new project, delete the ArtifactoryProject object from your namespace.
> * If you requested an increased quota, change the quota in the spec of your ArtifactoryProject back to the currentquota of your project. If you're not sure what the quota is, check the Overview page for your project in the Artifactory web GUI.
> 
> After you've made the change, Archeobot reconciles once more and you'll see that your `approval_status` changes back to `nothing-to-approve` (or you will have deleted the ArtifactoryProject object, in which case you won't see anything at all). This means your rejection has been acknowledged and you can make further change requests.

Reviewer: ```suggestion
```

This section should be deleted since I moved all of it up above the other stuff.

### Add a repository to your project

To add a repository to your project, do the following:
1. Click the gear icon and then choose **Repositories**. Click **Add Repositories**.
2. Choose a local repository. If you only need to push images to Artifactory, use a local repository.
3. Choose a package type.
4. Create a name for your new repository. All repositories in projects have the project key prefixed automatically. Use the following naming convention:

  `[desc]-[pkgtype]-[location]`

  For example, if you want to make a general-use docker repository for your team to push images to Artifactory, you could call your repository `gen-docker-local`.

5. Choose your environments. These tags affect user access. For example, you can grant allow a user to pull from all repos with the `dev` tag but not the `prod` tag. Make sure you grant the appropriate level of access to the users who need the artifacts in the repository.
6. Enable Xray indexing. Enable other options, if needed. Most of them can be left as their defaults.
8. Save the changes.

> Your new repository is set up and you can grant access to your service accounts.

Reviewer: ```suggestion
Your new repository is set up. If you would like your Artifactory Service Accounts to push to or pull from your new repository, please ensure you follow the instructions in the previous section to add your service account to the project.
```
